Mathlib/Topology/UnitInterval.lean

```diff
@@ -73,6 +72,13 @@ instance hasOne : One I :=
   ⟨⟨1, by constructor <;> norm_num⟩⟩
 #align unit_interval.has_one unitInterval.hasOne
 
+instance : ZeroLEOneClass I := ⟨@zero_le_one ℝ _ _ _ _⟩
+
+instance : BoundedOrder I := Set.Icc.boundedOrder zero_le_one
+
+lemma univ_eq_Icc : (univ : Set I) = Icc 0 1 := by
+  ext ⟨t, t0, t1⟩; simp_rw [mem_univ, true_iff]; exact ⟨t0, t1⟩
```

Comment:
This lemma looks a bit misleading! Perhaps to help the reader:
```suggestion
lemma univ_eq_Icc : (univ : Set I) = Icc (0 : I) (1 : I) := by
  ext ⟨t, t0, t1⟩; simp_rw [mem_univ, true_iff]; exact ⟨t0, t1⟩
```
It would be nice to have another lemma via `''` where the RHS lives in `ℝ`, but you can declare that out of scope

Reply:
Do you mean [Set.image_subtype_val_Icc](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Intervals/OrdConnected.html#Set.image_subtype_val_Icc)? (Apparently we lack the OrdConnected instance on Icc though.) That lemma would not be specific to unit interval; not every interval has Zero and One.

Reply:
I think it would follow from that lemma, yes; so `Subtype.val '' (univ : Set I) = Icc 0 1`.

Reply:
Oh I shouldn't be mentioning [Set.image_subtype_val_Icc](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Intervals/OrdConnected.html#Set.image_subtype_val_Icc). Your lemma is just [Set.image_univ](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Set.image_univ) trans [Subtype.range_val](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Subtype.range_val) (note `I` is abbrev of `Icc 0 1`).